gateway/api_loader.go

```diff
@@ -422,7 +421,9 @@ func processSpec(spec *APISpec, apisByListen map[string]int,
 			chainArray = append(chainArray, createDynamicMiddleware(obj.Name, false, obj.RequireSession, baseMid))
 		}
 	}
-
+	//Do not add middlewares after cache middleware.
+	//It will not get executed
+	mwAppendEnabled(&chainArray, &RedisCacheMiddleware{BaseMiddleware: baseMid, CacheStore: &cacheStore})
```

Comment:
AFAIK, `VirtualEndpoint` was working. Is there a specific case or something?

Reply:
I have no idea about `VirtualEndpoint` you are mentioning, however this make sense to me. As it ensures caching is the last middleware in the chain.